src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateProjectDependenciesRule.cs

```diff
@@ -10,39 +11,134 @@
 using Microsoft.DotNet.ProjectModel.Compilation;
 using Microsoft.DotNet.ProjectModel.Graph;
 using Microsoft.DotNet.Tools.Common;
+using NuGet.Frameworks;
 
 namespace Microsoft.DotNet.ProjectJsonMigration.Rules
 {
     public class MigrateProjectDependenciesRule : IMigrationRule
     {
         private readonly ITransformApplicator _transformApplicator;
+        private readonly  ProjectDependencyFinder _projectDependencyFinder;
         private string _projectDirectory;
 
         public MigrateProjectDependenciesRule(ITransformApplicator transformApplicator = null)
         {
             _transformApplicator = transformApplicator ?? new TransformApplicator();
+            _projectDependencyFinder = new ProjectDependencyFinder();
         }
 
         public void Apply(MigrationSettings migrationSettings, MigrationRuleInputs migrationRuleInputs)
         {
             _projectDirectory = migrationSettings.ProjectDirectory;
 
-            var csproj = migrationRuleInputs.OutputMSBuildProject;
-            var projectContext = migrationRuleInputs.DefaultProjectContext;
-            var projectExports = projectContext.CreateExporter("_").GetDependencies(LibraryType.Project);
+            var possibleProjectDependencies = _projectDependencyFinder
+                .FindPossibleProjectDependencies(migrationRuleInputs.DefaultProjectContext.ProjectFile.ProjectFilePath);
 
-            var projectDependencyTransformResults =
-                projectExports.Select(projectExport => ProjectDependencyTransform.Transform(projectExport));
+            var migratedXProjDependencyPaths = MigrateXProjProjectDependencies(migrationSettings, migrationRuleInputs);
+            var migratedXProjDependencyNames = migratedXProjDependencyPaths.Select(p => Path.GetFileNameWithoutExtension(p));
 
-            if (projectDependencyTransformResults.Any())
+
+            AddPropertyTransformsToCommonPropertyGroup(migrationRuleInputs.CommonPropertyGroup);
+            MigrateProjectJsonProjectDependencies(
+                possibleProjectDependencies, 
+                migrationRuleInputs.ProjectContexts, 
+                migratedXProjDependencyNames, 
+                migrationRuleInputs.OutputMSBuildProject);
+        }
+
+        private void ThrowIfUnresolvedDependencies(IEnumerable<ProjectContext> projectContexts, List<ProjectDependency> projectDependencies, IEnumerable<string> migratedXProjDependencyNames)
+        {
+            foreach (var projectContext in projectContexts)
             {
-                AddPropertyTransformsToCommonPropertyGroup(migrationRuleInputs.CommonPropertyGroup);
-                AddProjectDependenciesToNewItemGroup(csproj.AddItemGroup(), projectDependencyTransformResults);
+                var projectExports = projectContext.CreateExporter("_").GetDependencies(LibraryType.Project);
```

Comment:
`projectExports` isn't used.

This method is named "ThrowIf"... I assume GetDependencies does the throwing?